### PR TITLE
Documentation update

### LINKED ISSUE #600: Create documentation for the row by row function.

### PR DESCRIPTION
**Pull request by MichaelFurey17**
Close #600 #581 #609

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,44 +1,95 @@
...
 
 ## Repository Use Cases:
 - Enable business analysts to test the 2018 HMDA Platform
 - Provide files to check API returns
 - Provide a resource for developers to use while implementing business rules in code
 - Provide synthetic data files meeting the 2018 HMDA schema to be used in training
 
+## Repository Structure
+- [Python](https://github.com/cfpb/hmda-test-files/tree/master/python):
+    - Contains the notebooks and python scripts used to generate synthetic LAR data, perturb the data so that it fails specific business rules, and code to check that the produced files fail as expected.
```
Comment (Kibrael) [inline readme.md]: Can you add "edits" after `produced files fail` ?

File: readme.md [hunk 2]
```diff
@@ -1,44 +1,95 @@
...
+
+- [Schemas](https://github.com/cfpb/hmda-test-files/tree/master/schemas)
+    - Contains JSON objects that represent the data structures for LAR and Transmittal Sheet as defined by the [2018 HMDA FIG](https://s3.amazonaws.com/cfpb-hmda-public/prod/help/2018-hmda-fig-2018-hmda-rule.pdf)
+
+- [Dependencies](https://github.com/cfpb/hmda-test-files/tree/master/dependencies)
+    - Contains files used in the generation of synthetic LAR data.
+        - Tract to CBSA data for 2015 (most current year)
```
Comment (Kibrael) [inline readme.md]: This data was updated a few times, ours is old and needs an update here.
Can you specify what is currently being used and link to the file?

File: readme.md [hunk 3]
```diff
@@ -1,44 +1,95 @@
...
 - Provide files to check API returns
 - Provide a resource for developers to use while implementing business rules in code
 - Provide synthetic data files meeting the 2018 HMDA schema to be used in training
 
+## Repository Structure
+- [Python](https://github.com/cfpb/hmda-test-files/tree/master/python):
+    - Contains the notebooks and python scripts used to generate synthetic LAR data, perturb the data so that it fails specific business rules, and code to check that the produced files fail as expected.
+
+- [Schemas](https://github.com/cfpb/hmda-test-files/tree/master/schemas)
+    - Contains JSON objects that represent the data structures for LAR and Transmittal Sheet as defined by the [2018 HMDA FIG](https://s3.amazonaws.com/cfpb-hmda-public/prod/help/2018-hmda-fig-2018-hmda-rule.pdf)
```
Comment (Kibrael) [inline readme.md]: Excellent use of the current link for FIG :D

File: readme.md (hunk 1 above)
Comment (MichaelFurey17) [reply readme.md]: Sure! This change will be reflected in the next commit.

File: readme.md (hunk 3 above)
Comment (MichaelFurey17) [reply readme.md]: Thank you!

File: readme.md (hunk 2 above)
Comment (MichaelFurey17) [reply readme.md]: Sure! I can take out the parenthetical statement. If it is okay, we may want to add the new files and then update this documentation in another issue.

File: readme.md [hunk 4]
```diff
@@ -1,44 +1,95 @@
...
 
-## Generating Clean File
-Clean files will pass the HMDA edits (business rules for data submission). These files are used as the base for generating files that will fail edits.
-To generate clean files:
-- `python generate_clean_files.py`: running [this script](https://github.com/cfpb/hmda-test-files/blob/master/python/generate_clean_files.py) will create the edits_files directory and a data file that will pass the HMDA edit checks. The file will have a number of rows set in the YAML [configuration file](https://github.com/cfpb/hmda-test-files/blob/master/python/config.yaml). Other variables, such as data ranges can also be set in this file.
+## The FileGenerator Class. 
+The `FileGenerator()` class is located in `file_generator.py` and instantiates without any inputs in the function call. The class uses data stored with yaml configuration files in the 
+`python/configurations/` directory. The following yaml configuration files can be altered to create HMDA submission test files.  
 
+-`test_filepaths.yaml` holds the filepath names for storing clean files, edit test files, log files, and edit reports as well as the paths to the TS and LAR schemas.
```
Comment (Kibrael) [inline readme.md]: Can you add hyperlinks to the filenames?

File: readme.md [hunk 5]
```diff
@@ -1,44 +1,95 @@
...
+### Generating Clean Files and Edit Test Files. 
+The function method `create_files(kind)` outputs a clean test file or a set of edit test files, with an input indicating the kind of file to be created. Currently, the two inputs are `"clean_file"` which creates a clean file and `"error_files"` which produces files that will fail conditions for each edit. Edit test files are written to the directory indicated in `test_filepaths.yaml` based on edit type, such as /edits_files/test_files/Bank1/syntax/s300.txt. The file name indicates which edit the file is designed to test. If an edit has multiple conditions, a file will be made for each condition in the format /edits_files/test_files/Bank1/syntax/s301_1.txt.
```
Comment (Kibrael) [inline readme.md]: Is this `two inputs` referring to `kind`?

File: readme.md [hunk 6]
```diff
@@ -1,44 +1,95 @@
...
-- [Schemas](https://github.com/Kibrael/2018_test_files/tree/master/schemas)
-    - Contains JSON objects that represent the data structures for LAR and Transmittal Sheet as defined by the [2018 HMDA FIG](https://www.consumerfinance.gov/data-research/hmda/static/for-filers/2018/2018-HMDA-FIG.pdf)
+### Changing Edit Logic and the Classes Used for File Creation 
+The FileGenerator Class relies on four other classes, each with its own role in file creation. 
 
-- [Dependencies](https://github.com/Kibrael/2018_test_files/tree/master/dependancies)
-    - Contains files used in the generation of synthetic LAR data.
-        - Tract to CBSA data for 2015 (most current year)
-        - A file containing a list of US ZIP codes
+-`lar_generator` in `lar_generator.py` creates raw TS and LAR rows, the values of which are then manipulated to create clean test files or edit test files.
```
Comment (Kibrael) [inline readme.md]: Can you hyperlink these files?

File: readme.md (hunk 5 above)
Comment (MichaelFurey17) [reply readme.md]: Hopefully clarified this in a new section referenced in the next commit!

File: readme.md (hunk 6 above)
Comment (MichaelFurey17) [reply readme.md]: Sure! These changes should be reflected in the next commit.

**Pull request merged**